diff --git a/dist/js-yaml.js b/dist/js-yaml.js
index 4cc0ddf6d2616a37e196cf1fa2a8454022d86cd8..096080de43cd4c42117db3f0db0144a6d619bc03 100644
--- a/dist/js-yaml.js
+++ b/dist/js-yaml.js
@@ -1118,7 +1118,9 @@
   var CHOMPING_KEEP  = 3;
 
 
-  var PATTERN_NON_PRINTABLE         = /[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x84\x86-\x9F\uFFFE\uFFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF]/;
+  /* Reverts to previous version for compatibility with nginx */
+  var PATTERN_NON_PRINTABLE         = /[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x84\x86-\x9F\uFFFE\uFFFF]/;
+
   var PATTERN_NON_ASCII_LINE_BREAKS = /[\x85\u2028\u2029]/;
   var PATTERN_FLOW_INDICATORS       = /[,\[\]\{\}]/;
   var PATTERN_TAG_HANDLE            = /^(?:!|!!|![a-z\-]+!)$/i;
diff --git a/dist/js-yaml.mjs b/dist/js-yaml.mjs
index be71cad18da90d26e66ce43d419880ad008d27e0..fa73e14f4da2e185e9627f7e1927ebe27ffe650b 100644
--- a/dist/js-yaml.mjs
+++ b/dist/js-yaml.mjs
@@ -1111,8 +1111,9 @@ var CHOMPING_CLIP  = 1;
 var CHOMPING_STRIP = 2;
 var CHOMPING_KEEP  = 3;
 
+/* Reverts to previous version for compatibility with nginx */
+var PATTERN_NON_PRINTABLE         = /[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x84\x86-\x9F\uFFFE\uFFFF]/;
 
-var PATTERN_NON_PRINTABLE         = /[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x84\x86-\x9F\uFFFE\uFFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF]/;
 var PATTERN_NON_ASCII_LINE_BREAKS = /[\x85\u2028\u2029]/;
 var PATTERN_FLOW_INDICATORS       = /[,\[\]\{\}]/;
 var PATTERN_TAG_HANDLE            = /^(?:!|!!|![a-z\-]+!)$/i;
diff --git a/lib/loader.js b/lib/loader.js
index 39f13f56100655d8d66613dd7582ce58999a8c86..e6ed04855e88dfe8184f16ca3235be887938a0c2 100644
--- a/lib/loader.js
+++ b/lib/loader.js
@@ -21,8 +21,9 @@ var CHOMPING_CLIP  = 1;
 var CHOMPING_STRIP = 2;
 var CHOMPING_KEEP  = 3;
 
+/* Reverts to previous version for compatibility with nginx */
+var PATTERN_NON_PRINTABLE         = /[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x84\x86-\x9F\uFFFE\uFFFF]/;
 
-var PATTERN_NON_PRINTABLE         = /[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x84\x86-\x9F\uFFFE\uFFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF]/;
 var PATTERN_NON_ASCII_LINE_BREAKS = /[\x85\u2028\u2029]/;
 var PATTERN_FLOW_INDICATORS       = /[,\[\]\{\}]/;
 var PATTERN_TAG_HANDLE            = /^(?:!|!!|![a-z\-]+!)$/i;
